app/javascript/file-upload.js

```diff
@@ -17,17 +18,15 @@ MOJFrontend.MultiFileUpload.prototype.uploadFile = function (file) {
         processData: false,
         contentType: false,
         success: $.proxy(function (response) {
-            if (response.error) {
-                item.find('.moj-multi-file-upload__message').html(this.getErrorHtml(response.error));
-                this.status.html(response.error.message);
-            } else {
-                item.find('.moj-multi-file-upload__message').html(this.getSuccessHtml(response.success));
-                this.status.html(response.success.messageText);
-            }
+            feedback.html(this.getSuccessHtml(`${response.success.file_name} has been uploaded`));
+            this.status.html(`${response.success.file_name} has been uploaded`);
             item.find(`a.remove-link.moj-multi-file-upload__delete`).attr("value", response.success.evidence_id ?? file.name)
             this.params.uploadFileExitHook(this, file, response);
         }, this),
         error: $.proxy(function (jqXHR, textStatus, errorThrown) {
+            feedback.html(this.getErrorHtml(jqXHR.responseJSON.error.message));
+            this.status.html(jqXHR.responseJSON.error.message);
+            this.feedbackContainer.find('.govuk-table__row.moj-multi-file-upload__row').last().remove();
```

Comment:
I don't like this much as it assumes synchronous responses from the server... would prefer to remove by ID or some other identifier if possible... :)